### PR TITLE
Prompt fixes

### PR DESCRIPTION
**Pull request by dirkmueller**

### CODE REVIEW

File: chef/cookbooks/provisioner/templates/default/zzz-prompt.sh.erb [hunk 1]
```diff
@@ -1,31 +1,38 @@
 if test "$UID" = 0; then
-  prompt_suffix="#"
+    prompt_suffix="#"
+
```
Comment (vuntz) [inline chef/cookbooks/provisioner/templates/default/zzz-prompt.sh.erb]: Can you remove the new line here?

File: chef/cookbooks/provisioner/recipes/base.rb [hunk 2]
```diff
@@ -351,33 +351,33 @@
...
-        node["provisioner"]["shell_prompt"].to_s \
-          .gsub("USER", "\\[\\e[01;31m\\]\\u\\[\\e[0m\\]") \
-          .gsub("CWD", "\\[\\e[01;31m\\]\\w\\[\\e[0m\\]") \
-          .gsub("SUFFIX", "${prompt_suffix}") \
-          .gsub("ALIAS", "\\[\\e[01;35m\\]#{aliaz}\\[\\e[0m\\]") \
-          .gsub("HOST", "\\[\\e[01;35m\\]#{node["hostname"]}\\[\\e[0m\\]") \
-          .gsub("FQDN", "\\[\\e[01;35m\\]#{node["fqdn"]}\\[\\e[0m\\]")
+        node["provisioner"]["shell_prompt"].to_s. \
+          gsub("USER", "\\[\\e[01;35m\\]\\u\\[\\e[0m\\]"). \
+          gsub("CWD", "\\[\\e[01;35m\\]\\w\\[\\e[0m\\]"). \
```
Comment (bmwiedemann) [inline chef/cookbooks/provisioner/recipes/base.rb]: I wonder if there was a reason why cwd has the same color as user in bash, but a different one in zsh. Added in cc726563 by @tboerger

File: chef/cookbooks/provisioner/templates/default/zzz-prompt.sh.erb [hunk 3]
```diff
@@ -4,28 +4,34 @@ else
     prompt_suffix=">"
 fi
 
-case $(readlink /proc/$$/exe 2>/dev/null) in
-    */zsh)
-        PS1=$'<%= @zsh_prompt_from_template.call %> '
-        ;;
+case "$TERM" in
+    xterm*|vte*|linux*)
```
Comment (aspiers) [inline chef/cookbooks/provisioner/templates/default/zzz-prompt.sh.erb]: There are plenty of non-dumb terminals which don't match that pattern, e.g. `screen*`, `rxvt*`.  Better to do terminal capability tests via `tput` or similar.

Comment (dirkmueller) [inline chef/cookbooks/provisioner/templates/default/zzz-prompt.sh.erb]: @aspiers great idea ,thanks! done now.

**Pull request merged**